src/parser/jobs/blm/modules/RotationWatchdog.tsx

```diff
@@ -137,13 +141,45 @@ class Cycle {
 		return Math.max(this.expectedDespairs - this.actualDespairs, 0)
 	}
 
+	public get extraF1s(): number {
+		return Math.max(this.casts.filter(cast => cast.ability.guid === ACTIONS.FIRE_I.id).length - 1, 0)
+	}
+
+	public get hardT3Count(): number {
+		return this.casts.filter(cast => cast.ability.overrideAction)
+			.filter(cast => cast.ability.overrideAction.id === ACTIONS.THUNDER_III_FALSE.id).length
+	}
+	public get extraT3s(): number {
+		if (this.firePhaseStartMP < MIN_MP_FOR_FULL_ROTATION) {
+			return this.hardT3Count
+		}
+		return Math.max(this.hardT3Count - 1, 0)
+	}
+
+	public get manafontBeforeDespair(): boolean {
+		return this.hasManafont && this.actualDespairs > 0 &&
+			this.casts.findIndex(cast => cast.ability.guid === ACTIONS.MANAFONT.id) <
+			this.casts.findIndex(cast => cast.ability.guid === ACTIONS.DESPAIR.id)
+	}
+
+	public get isMissingFire(): boolean {
+		return !this.casts.filter(cast => FIRE_SPELLS.includes(cast.ability.guid)).length
```

Comment:
Oh dang, I wish I knew that was a thing earlier. There's a bunch of places in here that I'm doing checks for 'rotation did or didn't contain [x]' that I can use .some(<predicate>) instead of .filter(<predicate>).length <comparison>. Definitely fixing.